### PR TITLE
Modeling of workflow execution activities for metatranscriptomes 

### PR DESCRIPTION
**Pull request by aclum**
This PR addresses some gaps in how metatranscriptome modeling is done.  It 
- reuses the existing [ReadQcAnalysisActivity](https://microbiomedata.github.io/nmdc-schema/ReadQcAnalysisActivity/) b/c there are already metatranscriptom readqc records in mongo. We can change this after we roll out berkeley by adding by adding a new class + a type migrator.
- adds a corresponding Database slot for class [MetatranscriptomeAssembly](https://microbiomedata.github.io/nmdc-schema/MetatranscriptomeAssembly/)
- adds a corresponding Database slot for class [MetatranscriptomeAnnotationActivity](https://microbiomedata.github.io/nmdc-schema/MetatranscriptomeAnnotationActivity/)
- adds a new Class and corresponding slot for MetatranscriptomeExpression
- removes [MetatranscriptomeActivity](https://microbiomedata.github.io/nmdc-schema/MetatranscriptomeActivity/) which is a class that corresponding to running the workflow end to end instead of as modular components. 
- will require changes to berkeley schema migrators since those WorkflowExeuction subclass names are hard-coded.

### CODE REVIEW

File: src/data/invalid/Database-metatranscriptome_workflow-invalidDatabase.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    has_output:
+     - nmdc:dobj-11-547rwq75
+     - nmdc:dobj-11-547rwq77
+metatranscriptome_expression_set:
+  - id: nmdc:wfmtex-11-23989.1
+    git_url: https://github.com/microbiomedata/metaT_ReadCounts
+    version: v0.0.1
+    started_at_time: '2024-07-08T14:48:51+00:00'
+    ended_at_time: '2024-07-08T14:48:51+00:00'
+    type: MetatranscriptomeExpressionAnlaysis
```
Comment (eecavanna) [inline src/data/invalid/Database-metatranscriptome_workflow-invalidDatabase.yaml:50]:
```suggestion
    type: MetatranscriptomeExpressionAnalysis
```

File: src/data/valid/Database-metatranscriptome_workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    has_output:
+     - nmdc:dobj-11-547rwq75
+     - nmdc:dobj-11-547rwq77
+metatranscriptome_expression_set:
+  - id: nmdc:wfmtex-11-23989.1
+    git_url: https://github.com/microbiomedata/metaT_ReadCounts
+    version: v0.0.1
+    started_at_time: '2024-07-08T14:48:51+00:00'
+    ended_at_time: '2024-07-08T14:48:51+00:00'
+    type: MetatranscriptomeExpressionAnlaysis
```
Comment (eecavanna) [inline src/data/valid/Database-metatranscriptome_workflow.yaml:49]:
```suggestion
    type: MetatranscriptomeExpressionAnalysis
```

File: src/schema/nmdc.yaml [hunk 3]
```diff
@@ -2884,12 +2890,12 @@ slots:
     description:
       This property links a database object to the set of metagenome sequencing
       activities.
-  metatranscriptome_activity_set:
+  metatranscriptome_expression_set:
     domain: Database # not inherited by mixin users
     mixins: object_set
-    range: MetatranscriptomeActivity
+    range: MetatranscriptomeExpressionAnlaysis
```
Comment (eecavanna) [inline src/schema/nmdc.yaml:2896]:
```suggestion
    range: MetatranscriptomeExpressionAnalysis
```

File: src/schema/workflow_execution_activity.yaml [hunk 4]
```diff
@@ -215,10 +215,10 @@ classes:
           syntax: "{id_nmdc_prefix}:wfmtan-{id_shoulder}-{id_blade}{id_version}$"
           interpolated: true
 
-  MetatranscriptomeActivity:
+  MetatranscriptomeExpressionAnlaysis:
```
Comment (eecavanna) [inline src/schema/workflow_execution_activity.yaml:218]:
```suggestion
  MetatranscriptomeExpressionAnalysis:
```

**Review by eecavanna — CHANGES_REQUESTED** [review]
I made suggestions about the spelling of "analysis" in 4 places.

I'll ask separately (either in a separate PR comment or on Slack) about the migrator requirements.

**Comment by eecavanna** [comment]
Here's a link to the "Berkeley" migrator that moves all documents from the "child collections" of `workflow_execution_set`, into the `workflow_execution_set` collection.

- [`partials/migrator_from_10_2_0_to_11_0_0_part_16.py`](https://github.com/microbiomedata/berkeley-schema-fy24/blob/f4d2b4c0ca36891b48d84b938733eac8320b71a7/nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_16.py#L35C1-L48C11)

The migrator currently considers the "child collections" to be:
```py
        "workflow_execution_set": [
            "metagenome_annotation_set",
            "metagenome_assembly_set",
            "metatranscriptome_assembly_set",    # <-- created in this PR
            "metatranscriptome_annotation_set",  # <-- created in this PR
            "metatranscriptome_analysis_set",
            "mags_set",
            "metagenome_sequencing_set",
            "read_qc_analysis_set",
            "read_based_taxonomy_analysis_set",
            "metabolomics_analysis_set",
            "metaproteomics_analysis_set",
            "nom_analysis_set",
        ],
```

Highlights:
1. It already considers `metatranscriptome_assembly_set` and `metatranscriptome_annotation_set` to be "child collections" of the `workflow_execution_set` collection
2. It already does not consider `metatranscriptome_activity_set` to be a "child collection" of the `workflow_execution_set` collection
3. I does not yet consider `metatranscriptome_expression_set` to be a "child collection" of the `workflow_execution_set` collection

I think item 3 is the only place where the migrator falls short of accommodating this PR. Does that match your understanding, @aclum?

File: src/schema/nmdc.yaml [hunk 5]
```diff
@@ -2380,8 +2382,12 @@ enums:
         description: A failure has occurred in library preparation, a lab process.
       MetagenomeAssembly:
         description: A failure has occurred in metagenome assembly, a workflow process.
-      MetatranscriptomeActivity:
-        description: A failure has occurred in metatranscriptome analysis, a workflow process.
+      MetatranscriptomeExpressionActivity:
```
Comment (mbthornton-lbl) [inline src/schema/nmdc.yaml:2385]:
```suggestion
      MetatranscriptomeExpressionAnalysis:
```

File: src/schema/nmdc.yaml [hunk 6]
```diff
@@ -2939,6 +2945,17 @@ slots:
     mixins: object_set
     range: LibraryPreparation
     description: This property links a database object to the set of DNA extractions within it.
+  metatranscriptome_assembly_set:
```
Comment (mbthornton-lbl) [inline src/schema/nmdc.yaml:2948]: Is it possible to be more consistent with set_name and type here:
- using `Activity` in the type name(s) in a consistent way
- consistency between type and set_name in a CamelCase -> snake_case manner

**Review by mbthornton-lbl — DISMISSED** [review]
Suggested change in the comments.  Ideally we should be consistent with Type names - i.e. the inclusion of `Activity` or not, and consistency between Type and set_name

File: src/schema/nmdc.yaml (hunk 6 above)
Comment (aclum) [reply src/schema/nmdc.yaml:2948]: Im not using Activity for new classes to minimize changes needed for a backmerge with berkeley.

**Comment by turbomam** [comment]
@aclum or others: do you need help with the failed build process?

**Comment by aclum** [comment]
@turbomam No, I was able to get this to build. This is ready for re-review after I incorporated everyones analysis.

@eecavanna yes, what you've documented for what you'd need to change in the berkeley migrators is correct with excpetion of the minor change of metatranscriptome_expression_set to metatranscriptome_expression_analysis_set

**Review by turbomam — APPROVED** [review]

**Review by eecavanna — APPROVED** [review]
I'll create a ticket about updating the associated migrator.

**Pull request merged**